### PR TITLE
✨Enable shadow-demo environment for e2e tests for documents inside PWAs

### PR DESCRIPTION
**Pull request by cvializ**
Split off from #20994

### CODE REVIEW

File: build-system/tasks/e2e/amp-driver.js [hunk 1]
```diff
@@ -43,6 +44,21 @@ const EnvironmentBehaviorMap = {
       return `http://localhost:8000/examples/viewer.html#href=${url}`;
     },
   },
+
+  [AmpdocEnvironment.SHADOW_DEMO]: {
+    async ready(controller) {
+      // TODO(cvializ): this is a HACK
+      // There should be a better way to detect that the shadowdoc is ready.
+      const shadowHost = await controller.findElement(
+          '.amp-doc-host[style="visibility: visible;"]');
```
Comment (sparhami) [inline build-system/tasks/e2e/amp-driver.js]: Would do `'.amp-doc-host[style*="visibility: visible;"]'` to be robust against other styles being added in the future.

File: build-system/tasks/e2e/driver/query-xpath.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    // is not.
+    // It's ok to ignore this error since we just need the DOM structure.
+  }
+
+  // Map the xpath results from the fake tree to the corresponding nodes in
+  // the test document tree.
+  const xpathIterator = createXpathIterator(
+      evaluate(xpathString, fakeDocument, null, XPathResult.ANY_TYPE));
+  const elements = [...xpathIterator].map(node => {
+    const testId = node.dataset[TEST_ID_PROPERTY];
```
Comment (sparhami) [inline build-system/tasks/e2e/driver/query-xpath.js]: How about changing to use a  `getData` function to match the `setData` and `removeData` functions.

File: build-system/tasks/e2e/amp-driver.js (hunk 1 above)
Comment (cvializ) [reply build-system/tasks/e2e/amp-driver.js]: no : ( I'll add a more robust mechanism in a followup PR

File: build-system/tasks/e2e/driver/query-xpath.js (hunk 2 above)
Comment (cvializ) [reply build-system/tasks/e2e/driver/query-xpath.js]: good idea, I'll add this in a followup PR as well

Comment (cvializ) [reply build-system/tasks/e2e/driver/query-xpath.js]: merge conflict -_- I'll do this one now

**Pull request merged**